res/controllers/Pioneer-DDJ-SB-scripts.js

```diff
@@ -0,0 +1,888 @@
+};
+
+PioneerDDJSB.bindNonDeckControlConnections = function (isUnbinding) {
+    var samplerIndex,
+        fxUnitIndex;
+
+    for (samplerIndex = 1; samplerIndex <= 4; samplerIndex++) {
+        PioneerDDJSB.bindSamplerControlConnections('[Sampler' + samplerIndex + ']', isUnbinding);
+    }
+
+    for (fxUnitIndex = 1; fxUnitIndex <= 2; fxUnitIndex++) {
+        PioneerDDJSB.bindFXControlConnections('[EffectRack1_EffectUnit' + fxUnitIndex + ']', isUnbinding);
+    }
+};
+
+PioneerDDJSB.bindAllControlConnections = function (isUnbinding) {
+    var samplerIndex,
+        fxUnitIndex,
+        channelIndex;
+
+    for (samplerIndex = 1; samplerIndex <= 4; samplerIndex++) {
+        PioneerDDJSB.bindSamplerControlConnections('[Sampler' + samplerIndex + ']', isUnbinding);
+    }
+
+    for (fxUnitIndex = 1; fxUnitIndex <= 2; fxUnitIndex++) {
+        PioneerDDJSB.bindFXControlConnections('[EffectRack1_EffectUnit' + fxUnitIndex + ']', isUnbinding);
+    }
+
+    for (channelIndex = 1; channelIndex <= 2; channelIndex++) {
+        PioneerDDJSB.bindDeckControlConnections('[Channel' + channelIndex + ']', isUnbinding);
+    }
+};
+
+PioneerDDJSB.setDeckSoftTakeover = function (channel, isUnbinding) {
+    engine.softTakeover(channel, "volume", !isUnbinding);
+    engine.softTakeover(channel, "rate", !isUnbinding);
+    engine.softTakeover(channel, "pregain", !isUnbinding);
+    engine.softTakeover(channel, "filterHigh", !isUnbinding);
+    engine.softTakeover(channel, "filterMid", !isUnbinding);
+    engine.softTakeover(channel, "filterLow", !isUnbinding);
+    engine.softTakeover("[QuickEffectRack1_" + channel + "]", "super1", !isUnbinding);
+};
+
+PioneerDDJSB.setAllSoftTakeover = function (isUnbinding) {
+    var channelIndex;
+    for (channelIndex = 1; channelIndex <= 4; channelIndex++) {
+        PioneerDDJSB.setDeckSoftTakeover('[Channel' + channelIndex + ']', isUnbinding);
+    }
+};
+
+
+///////////////////////////////////////////////////////////////
+//                       DECK SWITCHING                      //
+///////////////////////////////////////////////////////////////
+
+PioneerDDJSB.deckSwitchTable = {
+    '[Channel1]': '[Channel1]',
+    '[Channel2]': '[Channel2]'
+};
+
+PioneerDDJSB.channelRegEx = /\[Channel(\d+)\]/;
+
+PioneerDDJSB.deckToggleButton = function (channel, control, value, status, group) {
+    var deckNumber = parseInt(PioneerDDJSB.channelRegEx.exec(PioneerDDJSB.deckSwitchTable[group])[1], 10);
+
+    if (value) {
+        if (deckNumber <= 2) {
+            deckNumber += 2;
+        } else {
+            deckNumber -= 2;
+        }
+        PioneerDDJSB.deckSwitchTable[group] = '[Channel' + deckNumber + ']';
+        PioneerDDJSB.initDeck(PioneerDDJSB.deckSwitchTable[group], false);
+        PioneerDDJSB.setDeckSoftTakeover(PioneerDDJSB.deckSwitchTable[group], true);
+        PioneerDDJSB.setDeckSoftTakeover(PioneerDDJSB.deckSwitchTable[group]);
```

Comment:
It is the only way I got soft takeover to work. If this is not done like this, soft takeover does not work properly. The first time I switch decks it works, but when switching back the control values change abruptly as if soft takeover was disabled. I don't know if this is the desired behavior. For me it seems like a bug in Mixxx.

The effects need even a worse looking workaround, probably because of the same issue (but turning soft takeover off and on again did not make the trick there for some reason).